### PR TITLE
add and change methods for services, modify collection form

### PR DESCRIPTION
**Pull request by janicka1**
Feature #14278

Arvados-DCO-1.1-Signed-off-by: Janicki Artur <artur.janicki@contractors.roche.com>

### CODE REVIEW

File: src/services/collection-service/collection-service.ts [hunk 1]
```diff
@@ -68,4 +69,29 @@ export class CollectionService extends TrashableResourceService<CollectionResour
         return this.webdavClient.put(fileURL, fileContent, requestConfig);
 
     }
+
+    update(uuid: string, data: Partial<CollectionResource>) {
+        if (uuid && data) {
```
Comment (unodgs) [inline src/services/collection-service/collection-service.ts]: why not simply `if (uuid && data && data.properties)` ?

File: src/services/groups-service/groups-service.ts [hunk 2]
```diff
@@ -37,20 +37,36 @@ export class GroupsService<T extends GroupResource = GroupResource> extends Tras
...
             filters: filters ? `[${filters}]` : undefined,
             order: order ? order : undefined
         };
-        return CommonResourceService.defaultResponse(
+
+        const response = await CommonResourceService.defaultResponse(
             this.serverApi
                 .get(this.resourceType + `${uuid}/contents`, {
-                    params: CommonResourceService.mapKeys(_.snakeCase)(params)
+                    params: CommonResourceService.mapKeys(_.camelCase)(params)
```
Comment (unodgs) [inline src/services/groups-service/groups-service.ts]: Isn't that a mistake? backend params are snake cased

File: src/services/groups-service/groups-service.ts [hunk 3]
```diff
@@ -37,20 +37,36 @@ export class GroupsService<T extends GroupResource = GroupResource> extends Tras
...
-            this.actions
+            this.actions, 
+            false
         );
+        const { items, ...res } = response;
+        const mappedItems = items.map((item: any) => {
+            if (item.kind === ResourceKind.COLLECTION) {
+                const { properties } = item;
+                return {
+                    ...TrashableResourceService.mapKeys(_.camelCase)(item),
```
Comment (unodgs) [inline src/services/groups-service/groups-service.ts]: That could be moved before `if` and reused

File: src/services/groups-service/groups-service.ts [hunk 4]
```diff
@@ -37,20 +37,34 @@ export class GroupsService<T extends GroupResource = GroupResource> extends Tras
...
             this.serverApi
                 .get(this.resourceType + `${uuid}/contents`, {
                     params: CommonResourceService.mapKeys(_.snakeCase)(params)
                 }),
-            this.actions
+            this.actions, 
+            false
         );
+        const { items, ...res } = response;
+        const mappedItems = items.map((item: any) => {
```
Comment (burzomir) [inline src/services/groups-service/groups-service.ts]: Avoid `any`. I think item type is inferred from items array and there is no need to add type info here.

**Pull request merged**